contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,350 @@
 //
 ////////////////////////////////////////////////////////////////////////////////
 #include "contrib/endpoints/src/api_manager/check_security_rules.h"
+#include <iostream>
+#include <sstream>
+#include "contrib/endpoints/src/api_manager/auth/lib/json_util.h"
+#include "contrib/endpoints/src/api_manager/proto/security_rules.pb.h"
+#include "contrib/endpoints/src/api_manager/utils/marshalling.h"
 
-#include <string>
-
-#include "contrib/endpoints/include/api_manager/api_manager.h"
-#include "contrib/endpoints/include/api_manager/request.h"
-
+using ::google::api_manager::auth::GetProperty;
+using ::google::api_manager::auth::GetStringValue;
 using ::google::api_manager::utils::Status;
+using ::google::protobuf::Map;
+using ::google::protobuf::util::error::Code;
 
 namespace google {
 namespace api_manager {
-
 namespace {
 
 const char kFirebaseServerStaging[] =
     "https://staging-firebaserules.sandbox.googleapis.com/";
 
+const char kFirebaseService[] =
+    "/google.firebase.rules.v1.FirebaseRulesService";
+
+const char kFailedFirebaseReleaseFetch[] = "Failed to fetch Firebase Release";
+const char kFailedFirebaseTest[] = "Failed to execute Firebase Test";
+const char kInvalidResponse[] = "Invalid JSON response from Firebase Service";
+
 // An AuthzChecker object is created for every incoming request. It does
 // authorizaiton by calling Firebase Rules service.
 class AuthzChecker : public std::enable_shared_from_this<AuthzChecker> {
  public:
-  AuthzChecker(std::shared_ptr<context::RequestContext> context,
-               std::function<void(Status status)> continuation);
+  // Constructor
+  AuthzChecker(ApiManagerEnvInterface *env,
+               auth::ServiceAccountToken *sa_token);
 
-  void Check();
+  // Check for Authorization success or failure
+  void Check(std::shared_ptr<context::RequestContext> context,
+             std::function<void(Status status)> continuation);
 
  private:
-  // Helper function to send a http GET request.
-  void HttpFetch(const std::string &url, const std::string &request_body,
+  // Helper method that invokes the test firebase service api.
+  void FirebaseCheck(std::string &ruleset_id,
+                     std::shared_ptr<context::RequestContext> context,
+                     std::function<void(Status status)> continuation);
+
+  // Parse the respose for GET RELEASE API call
+  Status ParseReleaseResponse(const std::string &json_str,
+                              std::string *ruleset_id);
+
+  // Parses the response for the TEST API call
+  Status ParseTestResponse(std::shared_ptr<context::RequestContext> context,
```

Comment:
Yes. Wayne gave me this comment and I have already fixed this. in a later commit.